### DISCUSSION TITLE
How to transcode and play uploaded VoD files as HLS in Ant Media Server without broadcasting them.

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#4888] [FAQs] [answered]
I'm uploading a VoD file to Ant Media Server, but how do I convert it to a multi-bitrate HLS stream without broadcasting because streaming with multiple ABRs uses more CPU on the server?

**Comment by tremzo** [comment, marked as answer]
*To convert uploaded VOD to HLS with different bit rates, please follow the below steps:*

1. First, you need to download the VOD to HLS transcode script on your server with the below command
 `wget https://raw.githubusercontent.com/ant-media/Scripts/master/vod_transcode.sh`

2. After downloading, run the command *`chmod +x vod_transcode.sh`* to give this script execute access.

By default, the script transcodes to 240p, 480p, and 720p, and the default target directory is `/usr/local/antmedia/webapps/WebRTCAppEE/streams/ ` but you can alter these as needed.

3. Make the following changes to the *advanced setting* of your application :
*`"vodUploadFinishScript"="/script-directory-path/vod_transcode.sh"`*

4. Now, upload any VOD file into the application where you used the script, and it will transcode it to HLS inside your target directory after execution.

5. Under the target directory, there will be master m3u8 and m3u8 files of different resolutions will be created which you can play with below URL format.
https://domain:5443/app-name/target-directory/Vod_Id.m3u8

**To learn more about the user-defined scripts for MP4 mixing and VoD upload, and detailed information please visit** https://antmedia.io/docs/guides/recording-live-streams/user-defined-scripts/#vod-upload-finish-process
